### PR TITLE
User actor-refacotring changes.

### PR DESCRIPTION
**Pull request by amolnin**

### CODE REVIEW

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                    StringFormatter.joinByComma(JsonKey.EXTERNAL_ID, JsonKey.EXTERNAL_ID_TYPE),
+                    JsonKey.EXTERNAL_ID_PROVIDER)),
+            ResponseCode.CLIENT_ERROR.getResponseCode());
+      }
+    }
+    return userId;
+  }
+
+  @SuppressWarnings({"unchecked"})
+  public static Map<String, Object> getUserFromExternalId(
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: Define this in UserExternalIdentityDao.getUserByExternalId.

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            (List<Map<String, Object>>) clientResponse.getResult().get(JsonKey.RESPONSE);
+        response = dataList.get(0);
+      }
+    } catch (Exception e) {
+      ProjectLogger.log("Validating client token failed due to : ", e);
+    }
+    return response;
+  }
+
+  public static Map<String, Object> getUserDetail(String userId) {
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: Define this in UserDao.getUserById

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            (List<Map<String, Object>>) userResponse.getResult().get(JsonKey.RESPONSE);
+        response = dataList.get(0);
+      }
+    } catch (Exception e) {
+      ProjectLogger.log("fetching user for id " + userId + " failed due to : ", e);
+    }
+    return response;
+  }
+
+  public static Map<String, Object> getOrgDetail(String orgId) {
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: Use OrgClient.getOrgById added by Rajat.

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      return encryptionService.encryptData(value);
+    } catch (Exception e) {
+      throw new ProjectCommonException(
+          ResponseCode.userDataEncryptionError.getErrorCode(),
+          ResponseCode.userDataEncryptionError.getErrorMessage(),
+          ResponseCode.SERVER_ERROR.getResponseCode());
+    }
+  }
+
+  public static Map<String, Object> getClientAccessTokenDetail(String clientId) {
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: Is this method used?

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+      if ((!StringUtils.isBlank(userId)) && (!userId.equals(ctxtUserId))) {
+        throw new ProjectCommonException(
+            ResponseCode.unAuthorized.getErrorCode(),
+            ResponseCode.unAuthorized.getErrorMessage(),
+            ResponseCode.UNAUTHORIZED.getResponseCode());
+      }
+    }
+  }
+
+  private static String getUserIdFromExtIdAndProvider(Request reqObj) {
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: Add this method in UserService.getUserId(Request request)

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+import org.sunbird.learner.util.Util;
+
+public class AuthenticationHelper {
+
+  private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private static EncryptionService encryptionService =
+      org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.getEncryptionServiceInstance(
+          null);
+
+  public static void validateWithUserId(Request reqObj) {
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: Add this in UserService.validateUserId(Request).

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.models.util.StringFormatter;
+import org.sunbird.common.models.util.datasecurity.EncryptionService;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.util.Util;
+
+public class AuthenticationHelper {
```
[the remark this answers is not in the document]
Comment (amolnin) [reply actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: class got removed completely

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java (hunk 1 above)
Comment (amolnin) [reply actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: This was also not used - so got removed along with class.

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java (hunk 2 above)
Comment (amolnin) [reply actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: Not used so deleted

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java (hunk 3 above)
Comment (amolnin) [reply actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: Deleted not used.

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java (hunk 4 above)
Comment (amolnin) [reply actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: Not used - got deleted. This methods were moved together from AuthenticationHelper - which was lying in Controller module

File: actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java (hunk 5 above)
Comment (amolnin) [reply actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java]: validateWithUserId - used from within this method - so not added to service layer.

File: actors/user/src/main/java/org/sunbird/user/dao/UserExternalIdentityDao.java [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package org.sunbird.user.dao;
+
+import org.sunbird.common.request.Request;
+
+public interface UserExternalIdentityDao {
+
+  public String getUserIdFromExtIdAndProvider(Request reqObj);
```
[the remark this answers is not in the document]
Comment (bvinayakumar) [reply actors/user/src/main/java/org/sunbird/user/dao/UserExternalIdentityDao.java]: Resolve codacy error.

File: actors/user/src/main/java/org/sunbird/user/service/UserService.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package org.sunbird.user.service;
+
+import org.sunbird.common.request.Request;
+
+public interface UserService {
+
+  public void validateWithUserId(Request reqObj);
```
[the remark this answers is not in the document]
Comment (bvinayakumar) [reply actors/user/src/main/java/org/sunbird/user/service/UserService.java]: Resolve codacy error.

File: actors/user/src/main/java/org/sunbird/user/service/impl/UserServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+import org.sunbird.user.dao.UserExternalIdentityDao;
+import org.sunbird.user.dao.impl.UserExternalIdentityDaoImpl;
+import org.sunbird.user.service.UserService;
+
+public class UserServiceImpl implements UserService {
+
+  private UserExternalIdentityDao userExtIdentityDao = new UserExternalIdentityDaoImpl();
+
+  @Override
+  public void validateWithUserId(Request reqObj) {
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/service/impl/UserServiceImpl.java]: validateWithUserId -> validateUserId

File: actors/user/src/main/java/org/sunbird/user/service/impl/UserServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    if (null != reqObj.getContext().get(JsonKey.IS_AUTH_REQ)
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/service/impl/UserServiceImpl.java]: IS_AUTH_REQ check not needed.

File: actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java [hunk 12]
```diff
@@ -60,21 +64,20 @@
...
+      "getUserDetailsByLoginId",
+      "profileVisibility",
+      "unblockUser",
+      "blockUser",
+      "assignRoles",
+      "userCurrentLogin",
+      "getMediaTypes"
+    },
+    asyncTasks = {})
 public class UserManagementActor extends BaseActor {
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java]: Another PR is required which splits UserManagementActor into UserActor, UserStatusActor, UserLoginActor, and UserProfileActor.

File: actors/user/src/main/java/org/sunbird/user/dao/impl/UserExternalIdentityDaoImpl.java [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+import org.sunbird.user.dao.UserExternalIdentityDao;
+
+public class UserExternalIdentityDaoImpl implements UserExternalIdentityDao {
+
+  private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private EncryptionService encryptionService =
+      org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.getEncryptionServiceInstance(
+          null);
+
+  public String getUserIdFromExtIdAndProvider(Request reqObj) {
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/dao/impl/UserExternalIdentityDaoImpl.java]: getUserByExternalId

File: actors/user/src/main/java/org/sunbird/user/dao/impl/UserExternalIdentityDaoImpl.java [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+                    StringFormatter.joinByComma(JsonKey.EXTERNAL_ID, JsonKey.EXTERNAL_ID_TYPE),
+                    JsonKey.EXTERNAL_ID_PROVIDER)),
+            ResponseCode.CLIENT_ERROR.getResponseCode());
+      }
+    }
+    return userId;
+  }
+
+  @SuppressWarnings({"unchecked"})
+  private Map<String, Object> getUserFromExternalId(String extId, String provider, String idType) {
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/dao/impl/UserExternalIdentityDaoImpl.java]: getUserByExternalId

**Pull request merged**